subprojects/enterprise/src/main/java/org/gradle/internal/tasks/TaskProperties.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.tasks;
+
+import java.util.Set;
+
+public interface TaskProperties {
```

Comment:
Why make this task-bound? I really-really hope that we will be able to capture artifact transforms and other work items soon. I think this is an opportunity to avoid adding more task-specific things, and start capturing more generic metadata. In practice I don't see why this couldn't be `WorkProperties` or `UnitOfWorkProperties`.

Reply:
Because we need it for test tasks. I have no objections if you want to implement it for more generic units of work. 🙂

Reply:
Another direction is to make it clear that this is only about TD, and bind it to the `Test` task specifically. In that case the type should be called `TestTaskProperties`.